File: Nueva carpeta/Lumen/benchmarks/c/bench.c
// Benchmark de referencia en C — mismos algoritmos que benchmarks/lumen/*.nv
#include <stdio.h>
#include <stdint.h>
#include <stdlib.h>
#include <string.h>

static long long fib(int n) {
    if (n < 2) return n;
    return fib(n - 1) + fib(n - 2);
}

static long long sum(long long n) {
    long long acc = 0;
    for (long long i = 0; i < n; i++) acc += i;
    return acc;
}

static int es_primo(long long n) {
    if (n < 2) return 0;
    if (n == 2) return 1;
    if (n % 2 == 0) return 0;
    for (long long i = 3; i * i <= n; i += 2)
        if (n % i == 0) return 0;
    return 1;
}

static long long primes(long long lim) {
    long long c = 0;
    for (long long k = 2; k < lim; k++)
        if (es_primo(k)) c++;
    return c;
}

static long long strings(long n) {
    long long total = 0;
    char buf[64];
    for (long i = 0; i < n; i++) {
        int len = snprintf(buf, sizeof buf, "item-%ld-fin", i);
        total += len;
    }
    return total;
}

static long long arrays(long n) {
    long long* xs = (long long*)malloc(sizeof(long long) * n);
    for (long i = 0; i < n; i++) xs[i] = i;
    long long acc = 0;
    for (long i = 0; i < n; i++) acc += xs[i];
    free(xs);
    return acc;
}

int main(int argc, char** argv) {
    if (argc < 2) { fprintf(stderr, "uso: bench <fib|sum|primes|strings|arrays>\n"); return 2; }
    const char* t = argv[1];
    if (!strcmp(t, "fib")) printf("fib:%lld\n", fib(26));
    else if (!strcmp(t, "sum")) printf("sum:%lld\n", sum(10000000));
    else if (!strcmp(t, "primes")) printf("primes:%lld\n", primes(20000));
    else if (!strcmp(t, "strings")) printf("strings:%lld\n", strings(200000));
    else if (!strcmp(t, "arrays")) printf("arrays:%lld\n", arrays(200000));
    else { fprintf(stderr, "tarea desconocida: %s\n", t); return 2; }
    return 0;
}

File: Nueva carpeta/Lumen/benchmarks/rust/bench.rs
// Benchmark de referencia en Rust — mismos algoritmos que benchmarks/lumen/*.nv
use std::env;

fn fib(n: i64) -> i64 {
    if n < 2 { n } else { fib(n - 1) + fib(n - 2) }
}

fn sum(n: i64) -> i64 {
    let mut acc: i64 = 0;
    let mut i: i64 = 0;
    while i < n {
        acc += i;
        i += 1;
    }
    acc
}

fn es_primo(n: i64) -> bool {
    if n < 2 { return false; }
    if n == 2 { return true; }
    if n % 2 == 0 { return false; }
    let mut i: i64 = 3;
    while i * i <= n {
        if n % i == 0 { return false; }
        i += 2;
    }
    true
}

fn primes(lim: i64) -> i64 {
    let mut c: i64 = 0;
    let mut k: i64 = 2;
    while k < lim {
        if es_primo(k) { c += 1; }
        k += 1;
    }
    c
}

fn strings(n: i64) -> i64 {
    let mut total: i64 = 0;
    let mut i: i64 = 0;
    while i < n {
        let s = format!("item-{}-fin", i);
        total += s.len() as i64;
        i += 1;
    }
    total
}

fn arrays(n: i64) -> i64 {
    let mut xs: Vec<i64> = Vec::with_capacity(n as usize);
    let mut i: i64 = 0;
    while i < n {
        xs.push(i);
        i += 1;
    }
    let mut acc: i64 = 0;
    let mut j: usize = 0;
    while j < xs.len() {
        acc += xs[j];
        j += 1;
    }
    acc
}

fn main() {
    let t = env::args().nth(1).expect("uso: bench <fib|sum|primes|strings|arrays>");
    match t.as_str() {
        "fib" => println!("fib:{}", fib(26)),
        "sum" => println!("sum:{}", sum(10_000_000)),
        "primes" => println!("primes:{}", primes(20_000)),
        "strings" => println!("strings:{}", strings(200_000)),
        "arrays" => println!("arrays:{}", arrays(200_000)),
        _ => panic!("tarea desconocida: {}", t),
    }
}
